### PR TITLE
Document the client layout properties

### PR DESCRIPTION
**Pull request by Elv13**
Hey! I got something done...

So this "finishes" the commits I was working on last March. I also added part of @Aire-One ongoing documentation work. I will take some more effort to "close the loop" on this and add the missing sections to @Aire-One page. I also originally planned to port the tag signal to the latest standards in the PR (the tag page is the last one using the old hardcoded signal doc instead of the ldoc based one).

Screenshot_20200809_234134

Screenshot_20200809_233859

The code isn't pretty, that whole template is kind of a mess. However it renders fine, for code we wont hopefully have to touch ever again, I think that's acceptable.

### CODE REVIEW

**Comment by Aire-One** [comment]
Yeay! I'm glad to see what you did here :)

I have to admit I'm kinda missing knowledge/time/motivation on my other PR (#2944)... So it makes me happy to see you're giving attention to this old stuff <3

**Comment by Elv13** [comment]
Ok, I rebased this pretty much as-is. I propose we merge it ASAP before we run out of Travis credits again

**Comment by Elv13** [comment]
@actionless , @Aire-One 

Are you fine if I merge this? I really want to do it before we run out of Travis CI credits again. We need the CI to update the website

File: tests/examples/wibox/awidget/defaults/taglist.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
+--DOC_GEN_IMAGEi --DOC_HIDE_ALL
```
Comment (Aire-One) [inline tests/examples/wibox/awidget/defaults/taglist.lua:1]:
```suggestion
--DOC_GEN_IMAGE --DOC_HIDE_ALL
```

File: tests/examples/wibox/awidget/defaults/watch.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
+--DOC_HIDE DOC_GEN_IMAGE --DOC_HEADER
```
Comment (Aire-One) [inline tests/examples/wibox/awidget/defaults/watch.lua:1]:
```suggestion
--DOC_HIDE --DOC_GEN_IMAGE --DOC_HEADER
```

**Review by Aire-One — APPROVED** [review]
I don't remember if I did a more in depth review last august...

Anyway, I took a quick overview of everything, and it looks good to me. I placed a few comments on some typos and I think I have a newer version for the `docs/08-client-layout-system.md` document. But it's not a big deal, and it can be fixed/improved later.

Let's merge it as-is! :rocket:

File: .travis.yml [hunk 3]
```diff
@@ -248,7 +248,10 @@ script:
       cd "$BUILD_IN_DIR"
       travis_run_in_fold "build_in_dir" cmake $CMAKE_ARGS "$SOURCE_DIRECTORY"
     fi
-  - travis_run_in_fold "make" make
+  - travis_run_in_fold "make" make ; sleep 5
+  - make ; sleep 5
+  - make ; sleep 5
+  - make ; sleep 5
```
Comment (blueyed) [inline .travis.yml:254]: @Elv13 
This ((1792780cf and 97a60818c)) shouldn't have been included here probably!?
Do you remember where it was coming from, and why the sleep might have been needed/useful?  (feel free to forward to Uli, of course :))

Comment (Elv13) [reply .travis.yml:254]: For a while now, some Travis builds "just stop" in the middle of printing stuff. This was an attempt to get the logs. Like this:

https://api.travis-ci.com/v3/job/492704164/log.txt

**Pull request merged**